components/Form/index.js

```diff
@@ -0,0 +1,332 @@
+import styled from "styled-components";
+import { css } from "styled-components";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+export default function Form({ onSubmit }) {
+  const [sleep, setSleep] = useState(true);
+  const [hygiene, setHygiene] = useState(false);
+  const [medicine, setMedicine] = useState(false);
+
+  const router = useRouter();
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    for (const property in data) {
+      if (data[property].trim() === "") {
+        return alert("Bitte gib einen Text in das Feld ein!");
+      }
+    }
+
+    onSubmit(
+      data.name,
+      data.category,
+      data.adress,
+      data.link,
+      data.target,
+      data.requirements,
+      data.filterCriteria,
+      data.tel
```

Comment:
You could destructure all different elements from `data`, so you do not have to repeat `data. ...` so many times :)